File: gymify-spa/projects/api-client/src/lib/clients/group-sessions/models/group-session.dto.ts
import { UUID } from '../../../types/uuid.type';

export interface GroupSessionDTO {
  groupSessionUid: UUID;
  groupSessionName: string;
  hour: number;
  place: string;
  coachName: string;
  duration: string;
  availableSlots: number;
  takenSlots: number;
  description: string;
  isBookedIn: boolean;
  canEdit: boolean;
  startDate: Date;
  endDate: Date;
  placeId: number;
}

File: Gymify.Application/GroupSessions/Queries/GetGroupSessions/GroupSessionDTO.cs
namespace Gymify.Application.GroupSessions.Queries.GetGroupSessions;

public class GroupSessionDTO
{
    public Guid GroupSessionUid { get; init; }
    public string Hour { get; init; }
    public string GroupSessionName { get; init; }
    public string Place { get; init; }
    public string CoachName { get; init; }
    public string Duration { get; init; }
    public int AvailableSlots { get; init; }
    public int TakenSlots { get; init; }
    public string Description { get; init; }
    public bool IsBookedIn { get; init; }
}